gtsfm/loader/yfcc_imb_loader.py

```diff
@@ -0,0 +1,190 @@
+"""Loader for the Image-Matching-Benchmark's YFCC dataset.
+
+References: https://www.cs.ubc.ca/research/image-matching-challenge/
+
+Authors: Ayush Baid
+"""
+import os.path as osp
+from typing import List, Optional
+
+import numpy as np
+from gtsam import Cal3Bundler, PinholeCameraCal3Bundler, Pose3, Rot3
+
+import gtsfm.utils.io as io_utils
+import thirdparty.colmap.read_write_model as colmap_read_write_model
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class YfccImbLoader(LoaderBase):
```

Comment:
is there a Github python file from the IMB Github we are basing some of this off of?

Reply:
Maybe we could add a ref to it, if there is

Reply:
@ayushbaid bumping this comment

Reply:
Thanks for this. I now notice there are actually two different places from which the camera pose information is read. I used this as a reference: https://github.com/vcg-uvic/image-matching-benchmark/blob/master/example/training_data/parse_data.ipynb. However, in the stereo computation, a different source is used: https://github.com/vcg-uvic/image-matching-benchmark/blob/d6953b32e07e3545ebd50ae52887cc7c91e433a1/compute_stereo.py#L77